### PR TITLE
chore: truncate versions for controller build that interfere with kubernetes 63 char limit

### PR DESCRIPTION
**Pull request by morri-son**
On-behalf-of: Gerald Morrison (SAP) <gerald.morrison@sap.com>

#### What this PR does / why we need it
introduce new helper for controller chart to truncate Chart version
  

## Summary by CodeRabbit

* **Bug Fixes**
  * Corrected monitoring TLS configuration so certificate key settings are properly aligned and used.

* **Chores**
  * Enforced a maximum length for generated version labels to meet platform constraints; overly long pseudo-versions are truncated and a warning is emitted when truncation occurs.

* **Tests**
  * Added tests covering version truncation behavior, warning emission, and boundary conditions.

### CODE REVIEW

**Review by jakobmoellerdev — DISMISSED** [review]

File: kubernetes/controller/chart/templates/_helpers.tpl [hunk 1]
```diff
@@ -31,6 +31,13 @@ Always uses the Helm release namespace.
 {{- .Release.Namespace }}
 {{- end }}
 
+{{/*
+Create chart label value (chart name + version), truncated to 63 characters.
+*/}}
+{{- define "ocm-k8s-toolkit.chart" -}}
```
Comment (matthiasbruns) [inline kubernetes/controller/chart/templates/_helpers.tpl]: I do not think we should truncate within the chart - users need to make sure they provide valid resource names, it's now our duty to clean this up for them - the fix should not be here but where we build the charts in the cli

**Review by matthiasbruns — CHANGES_REQUESTED** [review]
I think the fix is in the wrong place - the charts should not have value clean up - the values are provided by devops - they either know about the resource limit or will have to handle the errors themselves

the fix should be in the pipeline, where we build the controller and truncate the name there OR in create-version.js and truncate the output

**Review by matthiasbruns — DISMISSED** [review]

File: .github/scripts/compute-version.test.js [hunk 2]
```diff
@@ -201,3 +201,132 @@ assert.notStrictEqual(
...
 
 console.log("✅ All tests passed.");
+
+// ----------------------------------------------------------
+// Version truncation tests
+// ----------------------------------------------------------
+console.log("Testing version truncation...");
+
+// DEFAULT_MAX_VERSION_LENGTH should be exported and equal 57
+assert.strictEqual(
```
Comment (piotrjanik) [inline .github/scripts/compute-version.test.js]: This value is harcoded, so very little value of this test.

**Review by piotrjanik — APPROVED** [review]

**Review by jakobmoellerdev — APPROVED** [review]
